network/src/connectivity_manager/mod.rs

```diff
@@ -188,14 +195,25 @@ impl DiscoveredPeerSet {
     }
 }
 
-#[derive(Clone, Debug, Default, PartialEq, Serialize)]
+/// Represents all the information for a discovered peer
+#[derive(Clone, Debug, PartialEq, Serialize)]
 struct DiscoveredPeer {
     role: PeerRole,
     addrs: Addresses,
     keys: PublicKeys,
+    /// The last time the node was dialed
+    last_dial_time: SystemTime,
 }
 
 impl DiscoveredPeer {
+    pub fn new(role: PeerRole) -> Self {
+        Self {
+            role,
+            addrs: Addresses::default(),
+            keys: PublicKeys::default(),
+            last_dial_time: SystemTime::now(),
```

Comment:
This doesn't look right semantically as we have not dialed to the peer as of now?

Reply:
Fixed to epoch